dali/test/python/test_operator_decoders_image.py

```diff
@@ -115,26 +131,81 @@ def check(img_type, size, device, threads):
         with check_output_pattern(pattern):
             run_decode(data_path, size, device, threads, memory_stats=True)
 
-    for threads in {1, 2, 3, 4}:
-        for size in {1, 10}:
+    for size in {1, 10}:
+        for threads in {1, random.choice([2, 3, 4])}:
             yield check, img_type, size, device, threads
 
+batch_size_test = 16
+@pipeline_def(batch_size=batch_size_test, device_id=0, num_threads=4)
+def img_decoder_pipe(device, out_type, files):
+    encoded, _ = fn.readers.file(files=files)
+    decoded = fn.decoders.image(encoded, device=device, output_type=out_type)
+    return decoded
 
-
-batch_size_alias_test=16
-
-@pipeline_def(batch_size=batch_size_alias_test, device_id=0, num_threads=4)
+def _testimpl_image_decoder_consistency(img_out_type, file_fmt, path, subdir='*', ext=None):
+    eps = 1
+    if file_fmt == 'jpeg' or file_fmt == 'mixed':
+        eps = 4
+    if ((file_fmt == 'jpeg2k' or file_fmt == 'mixed') and img_out_type == types.YCbCr):
+        eps = 6
+    files = get_img_files(os.path.join(test_data_root, path), subdir=subdir, ext=ext)
+    compare_pipelines(img_decoder_pipe("cpu", out_type=img_out_type, files=files),
+                      img_decoder_pipe("mixed", out_type=img_out_type, files=files),
+                      batch_size=batch_size_test, N_iterations=3,
+                      eps=eps)
+
+def test_image_decoder_consistency():
+    for out_img_type in [types.RGB, types.BGR, types.YCbCr, types.GRAY, types.ANY_DATA]:
```

Comment:
How generic_image handles types.YCbCr ?

Reply:
there is a color conversion step at the end (OpenCV gives you BGR always)